### DISCUSSION TITLE
Cannot style megamenu label and panel label separately with pass through

### DISCUSSION THREAD

**Discussion by amod81** [primefaces/community#297] [PrimeVue] [answered]
I am trying to style the MegaMenu with PassThrough. I could change the background color, icon color, sub menu icon and the label on the menu (changed to white as in the attached image). However, when I apply a class to the label, it gets applied to the panel -> menu items as well. As a result, I cannot distinguish the label color on the menu and inside the panel (sub menus). Should these 2 elements be identified by different property names in future? How do I configure 2 different colors on labels of the main menu and those of the sub-menu-items.

The image below does not show the sub-menus under User 1 User 2 ... because their fore color has changed (to white) after I have applied the class to the labels "Users", "Events", "Settings".

![image](https://github.com/primefaces/community/assets/93420889/a63012bf-5ae8-43b3-b477-348b7aa9dd99)

Code

```
  <MegaMenu
      :model="items"
      :pt="{
          label: {
              class: ['text-white']
          },
          icon: {
              class: ['text-white']
          },
          submenuIcon: {
              class: ['text-white']
          },
          submenuheader: {
              class: ['text-primary']
          },
      }"
  />
```

```
label: {
  class: ['text-white']
},
```
gets applied to all 'label' of the megamenu and hence cannot differentiate main-menu.label and content/panel.label with different colors.

Environment:
    "primeflex": "^3.3.1",
    "primeicons": "^6.0.1",
    "primevue": "^3.34.1",

**Comment by tugcekucukoglu** [comment]
The use of label options for both root and panel items has caused confusion. I will change the ones on the panel to `submenulabel`.

Sorry for the confusion.

**Comment by tugcekucukoglu** [comment, marked as answer]
Fixed for the next version. https://github.com/primefaces/primevue/issues/4690

**Reply by amod81** [reply]
Brilliant !! Thank You!

**Comment by amod81** [comment]
Hi,

The same applies to the Menubar component as well. MegaMenu is all good now after the fix in 3.39.

Appreciate your help !!
